packages/lib/src/core/Analytics/Analytics.ts

```diff
@@ -1,74 +1,82 @@
-                this.checkoutAttemptId = this.props.checkoutAttemptId;
-                this.queue.run(this.checkoutAttemptId);
-            }
+    const { telemetry, enabled } = props;
+    if (telemetry === true && enabled === true) {
+        if (props.checkoutAttemptId) {
+            // handle prefilled checkoutAttemptId // TODO is this still something that ever happens?
+            _checkoutAttemptId = props.checkoutAttemptId;
         }
     }
 
-    send(event) {
-        const { enabled, payload, telemetry } = this.props;
+    const _logEvent = logEvent({ loadingContext, locale });
+    const _collectId = collectId({ analyticsContext, clientKey, locale, amount, analyticsPath: ANALYTICS_PATH });
+    const _caEventsQueue: EQObject = CAEventsQueue({ analyticsContext, clientKey, analyticsPath: ANALYTICS_PATH });
 
-        if (enabled === true) {
-            if (telemetry === true && !this.checkoutAttemptId) {
-                // fetch a new checkoutAttemptId if none is already available
-                this.collectId()
-                    .then(checkoutAttemptId => {
-                        this.checkoutAttemptId = checkoutAttemptId;
-                        this.queue.run(this.checkoutAttemptId);
-                    })
-                    .catch(e => {
-                        console.warn(`Fetching checkoutAttemptId failed.${e ? ` Error=${e}` : ''}`);
-                    });
+    const analyticsObj: AnalyticsModule = {
+        send: (initialEvent: AnalyticsInitialEvent) => {
+            const { enabled, payload, telemetry } = props; // TODO what is payload, is it ever used?
+
+            if (enabled === true) {
+                if (telemetry === true && !_checkoutAttemptId) {
+                    // fetch a new checkoutAttemptId if none is already available
+                    _collectId({ ...initialEvent, ...(payload && { ...payload }) })
+                        .then(checkoutAttemptId => {
+                            _checkoutAttemptId = checkoutAttemptId;
+                        })
+                        .catch(e => {
+                            // Caught at collectId level. We do not expect this catch block to ever fire, but... just in case...
+                            console.debug(`Fetching checkoutAttemptId failed.${e ? ` Error=${e}` : ''}`);
+                        });
+                }
+                // Log pixel // TODO once we stop using the pixel we can stop requiring both "enabled" & "telemetry" config options
+                _logEvent(initialEvent);
             }
+        },
 
-            if (telemetry === true) {
-                const telemetryTask = checkoutAttemptId =>
-                    this.logTelemetry({ ...event, ...(payload && { ...payload }), checkoutAttemptId }).catch(() => {});
+        getCheckoutAttemptId: (): string => _checkoutAttemptId,
 
-                this.queue.add(telemetryTask);
+        addAnalyticsAction: (type: ANALYTICS_ACTION, obj: AnalyticsObject) => {
```

Comment:
I think maybe we could simplify the interface a bit (later stage).

The idea is more like:
`Components` either dispatch custom events which will be captured by the `Analytics` or have the `Analytics` as the context. The only thing Components need to do is to send an event to the `Analytics` module.

`Analytics` is responsible to dump the events to `Queue` and fetch the `checkoutAttemptId` when missing (just some pre work before sending the message to the `Queue`).

`Queue` manages to dispatch the events to the correct queue type - the error queue or log queue.
When the error queue has a message, send it right away. When the log queue has messages, it will wait until it's full. The queue size is configurable.

I think this is more decoupled. The `Queue` can have unlimited types of queues and `Analytics` module doesn't care which type of queue to send. Furthermore, the `Analytics` module doesn't need to know how to `debounce`. 
The `Analytics` only needs to expose one public function to all components - `send`(or none if components just dispatch events.)

Reply:
Maybe a refactor for later _if_ we end up needing a more generic kind of queue. 
At the moment Analytics & EventsQueue only have one purpose - to access the analytics endpoint